dnn/POC_synthetic_segmentation/generation/scene_setup_transparent.py

```diff
@@ -0,0 +1,109 @@
+'''
+This script is to be used in combination with model files and resource files available at: 
+https://uillinoisedu-my.sharepoint.com/personal/yixiaos3_illinois_edu/_layouts/15/onedrive.aspx?originalPath=aHR0cHM6Ly91aWxsaW5vaXNlZHUtbXkuc2hhcmVwb2ludC5jb20vOmY6L2cvcGVyc29uYWwveWl4aWFvczNfaWxsaW5vaXNfZWR1L0VrUWVuVVhWaWVsTWt1a2p1M1dGcjVjQm1veHBGRFFlQUZZek1fZUpFSExBVXc_cnRpbWU9WnQ4TmY2WkgyRWc&id=%2Fpersonal%2Fyixiaos3_illinois_edu%2FDocuments%2FRM 2021%2FAI Vision%2FDataset%2FSynthesis%2FBatch_Render_Infantry
+'''
+import bpy
+import numpy as np
+C = bpy.context
+D = bpy.data
+seed=0
+np.random.seed(seed)
+print(f'seed: {seed}')
+
+# CHANGE ME
+DO_LABEL=True # If we are rendering label or image
+DO_NUMBER=False # If numbers are included
+NUM_SAMPLE=1e3 # How many images to generate
+
+def get_all_objs():
+    objs=[]
+    for collection in bpy.data.collections:
+       print(collection.name)
+       if not 'Exclude' in collection.name:
+            for obj in collection.all_objects:
+                objs.append(obj)
+    return objs
+
+def set_visible_all(objs,visibility=True):
+    visibility=not visibility
+    for obj in objs:
+        obj.hide_set(visibility)
+        obj.hide_render=visibility
+        obj.hide_viewport=visibility
+
+def update(p1,p2):
+    text = bpy.data.objects['t1']
+    frame = bpy.context.scene.frame_current
+    if DO_NUMBER:
+        text.data.body = str(frame%6+1)
+    else:
+        text.data.body=''
+
+
+# TODO Lighting, Camera Background
+def simulate(scene, robot_objects,label_objects,stage_objects,light_objects,number_objects,obj_camera,camera_target,label=False):
+    bpy.ops.object.select_all(action='DESELECT')
+    mesh_objs=label_objects+robot_objects+label_objects+stage_objects+light_objects+[obj_camera,camera_target]+number_objects
+    for ob in mesh_objs:
+        ob.animation_data_clear()
+        ob.select_set(state = True)
+    if label:
+        scene.render.filepath = bpy.data.filepath.rpartition('.')[0]+'/label_out/out_seed{}_'.format(seed)
+        obj_camera.data.show_background_images=False
+    else:
+        scene.render.filepath = bpy.data.filepath.rpartition('.')[0]+'/image_out/out_seed{}_'.format(seed)
+        obj_camera.data.show_background_images=True
+    set_visible_all(get_all_objs(),not label)
+    set_visible_all(label_objects,label)
+    
+    try: 
+        bpy.app.handlers.frame_change_post.remove(update)
+    except: pass
```

Comment:
What is this doing?

Reply:
Re-registering frame update callback with the newer update function in code. Useful when you modify the update function and rerun the script without restarting blender.

Reply:
Why do we need this `try` `except: pass` thing? The error handling looks a bit sketchy

Reply:
I agree. Changed logic to just clear all old callbacks instead.